tesseract/tesseract_planning/tesseract_process_managers/src/process_generators/fix_state_collision_process_generator.cpp

```diff
@@ -0,0 +1,275 @@
+  }
+  catch (std::runtime_error e)
+  {
+    CONSOLE_BRIDGE_logError("MoveWaypointFromCollision error: %s", e.what());
+    return false;
+  }
+  std::size_t num_jnts = static_cast<std::size_t>(start_pos.size());
+
+  pci.init_info.type = InitInfo::GIVEN_TRAJ;
+  pci.init_info.data = tesseract_common::TrajArray(2, start_pos.size());
+  pci.init_info.data.row(0) = start_pos.transpose();
+  pci.init_info.data.row(1) = start_pos.transpose();
+
+  // Populate Cost Info
+  auto jv = std::make_shared<JointPosTermInfo>();
+  jv->coeffs = std::vector<double>(num_jnts, 1.0);
+  jv->targets = std::vector<double>(start_pos.data(), start_pos.data() + start_pos.size());
+  jv->upper_tols = std::vector<double>(num_jnts, profile.joint_tolerance);
+  jv->lower_tols = std::vector<double>(num_jnts, -profile.joint_tolerance);
+  jv->first_step = 0;
+  jv->last_step = 1;
+  jv->name = "joint_pos";
+  jv->term_type = TT_COST;
+  pci.cost_infos.push_back(jv);
+
+  {
+  auto collision = std::make_shared<CollisionTermInfo>();
+  collision->name = "collision";
+  collision->term_type = TT_CNT;
+  collision->evaluator_type = trajopt::CollisionEvaluatorType::SINGLE_TIMESTEP;
+  collision->first_step = 0;
+  collision->last_step = 1;
+  collision->info = createSafetyMarginDataVector(pci.basic_info.n_steps, 0.025, 1);
+  collision->use_weighted_sum = true;
+  pci.cnt_infos.push_back(collision);
+  }
+  {
+  auto collision = std::make_shared<CollisionTermInfo>();
+  collision->name = "collision";
+  collision->term_type = TT_COST;
+  collision->evaluator_type = trajopt::CollisionEvaluatorType::SINGLE_TIMESTEP;
+  collision->first_step = 0;
+  collision->last_step = 1;
+  collision->info = createSafetyMarginDataVector(pci.basic_info.n_steps, 0.025, 20);
+  collision->use_weighted_sum = true;
+  pci.cost_infos.push_back(collision);
+  }
+  auto prob = ConstructProblem(pci);
+
+  // See if there are collisions
+  using namespace tesseract_collision;
+  using namespace tesseract_environment;
+
+    std::vector<ContactResultMap> collisions;
+    tesseract_environment::StateSolver::Ptr state_solver = prob->GetEnv()->getStateSolver();
+    DiscreteContactManager::Ptr manager = prob->GetEnv()->getDiscreteContactManager();
+    AdjacencyMap::Ptr adjacency_map =
+        std::make_shared<tesseract_environment::AdjacencyMap>(prob->GetEnv()->getSceneGraph(),
+                                                              prob->GetKin()->getActiveLinkNames(),
+                                                              prob->GetEnv()->getCurrentState()->link_transforms);
+
+    manager->setActiveCollisionObjects(adjacency_map->getActiveLinkNames());
+    manager->setContactDistanceThreshold(0);
+    collisions.clear();
+
+
+
+  // This never returns any collisions
+  if (!checkTrajectory(collisions, *manager, *state_solver, prob->GetKin()->getJointNames(), prob->GetInitTraj()))
+//    return true;
+    std::cout << "No collisions found."  << std::endl;
```

Comment:
@Levi-Armstrong Am I doing this wrong? It never returns any collisions.

The optimization always converges as well and the collision terms are 0.

Reply:
Just to check I would verify that the issue is not with the new state solver and switch it KDL. Are you saying that during optimization it never reports any collisions?

Reply:
Yeah. It returns with a constraint that is satisfied but the post check finds collisions. I was concerned that something might be wrong in longest valid segment code for instance since it is a trajectory of points in the same location.